packages/netlify-cms-backend-github/src/GraphQLAPI.js

```diff
@@ -0,0 +1,459 @@
+      return {
+        headers: {
+          ...headers,
+          authorization: this.token ? `token ${this.token}` : '',
+        },
+      };
+    });
+    const httpLink = createHttpLink({ uri: `${this.api_root}/graphql` });
+    return new ApolloClient({
+      link: authLink.concat(httpLink),
+      cache: new InMemoryCache({ fragmentMatcher }),
+      defaultOptions: {
+        watchQuery: {
+          fetchPolicy: NO_CACHE,
+          errorPolicy: 'ignore',
+        },
+        query: {
+          fetchPolicy: NO_CACHE,
+          errorPolicy: 'all',
+        },
+      },
+    });
+  }
+
+  reset() {
+    return this.client.resetStore();
+  }
+
+  async getRepository() {
+    const { repo_owner: owner, repo_name: name } = this;
+    const { data } = await this.query({
+      query: queries.repository,
+      variables: { owner, name },
+      fetchPolicy: CACHE_FIRST, // repository id doesn't change
+    });
+    return data.repository;
+  }
+
+  query(options = {}) {
+    return this.client.query(options).catch(error => {
+      throw new APIError(error.message, 500, 'GitHub');
+    });
+  }
+
+  mutate(options = {}) {
+    return this.client.mutate(options).catch(error => {
+      throw new APIError(error.message, 500, 'GitHub');
+    });
+  }
+
+  async hasWriteAccess() {
+    const { repo_owner: owner, repo_name: name } = this;
+    try {
+      const { data } = await this.query({
+        query: queries.repoPermission,
+        variables: { owner, name },
+        fetchPolicy: CACHE_FIRST, // we can assume permission doesn't change often
+      });
+      // https://developer.github.com/v4/enum/repositorypermission/
+      const { viewerPermission } = data.repository;
+      return ['ADMIN', 'MAINTAIN', 'WRITE'].includes(viewerPermission);
+    } catch (error) {
+      console.error('Problem fetching repo data from GitHub');
+      throw error;
+    }
+  }
+
+  async user() {
+    const { data } = await this.query({
+      query: queries.user,
+      fetchPolicy: CACHE_FIRST, // we can assume user details don't change often
+    });
+    return data.viewer;
+  }
+
+  async retrieveBlobObject(expression, options = {}) {
+    const { repo_owner: owner, repo_name: name } = this;
+    const { data } = await this.query({
+      query: queries.blob,
+      variables: { owner, name, expression },
+      ...options,
+    });
+    // https://developer.github.com/v4/object/blob/
+    if (data.repository.object) {
+      const { is_binary, text } = data.repository.object;
+      return { is_null: false, is_binary, text };
+    } else {
+      return { is_null: true };
+    }
+  }
+
+  async retrieveContent(path, branch) {
+    const { is_null, is_binary, text } = await this.retrieveBlobObject(`${branch}:${path}`);
+    if (is_null) {
+      throw new APIError('Not Found', 404, 'GitHub');
+    } else if (!is_binary) {
+      return text;
+    } else {
+      return super.retrieveContent(path, branch);
```

Comment:
The GraphQL docs are not really helpful on how to deal with binary data. They just mention that in case of binary data `text` is null and `is_binary` is true, so I decided to revert to the REST API in that case